src/main/java/com/aws/greengrass/deployment/ThingGroupHelper.java

```diff
@@ -102,4 +105,41 @@ public Optional<Set<String>> listThingGroupsForDevice(int maxAttemptCount) throw
             return Optional.of(thingGroupNames);
         }, "get-thing-group-hierarchy", logger);
     }
+
+    public void waitForReconnect(long timeoutMillis) throws Exception {
```

Comment:
a failure in the thing group helper does not prevent a deployment from working, I don't think you want to validate this.

Reply:
The idea here is to validate the http client can make a successful API call and thing group helper would be a quick and easy way to do that.